idea/idea-frontend-api/src/org/jetbrains/kotlin/idea/frontend/api/types/ktTypeUtils.kt

```diff
@@ -5,17 +5,66 @@
 
 package org.jetbrains.kotlin.idea.frontend.api.types
 
-import org.jetbrains.kotlin.builtins.KotlinBuiltIns
 import org.jetbrains.kotlin.builtins.StandardNames
 import org.jetbrains.kotlin.name.ClassId
 
+val KtType.isMarkedNullable: Boolean get() = (this as? KtTypeWithNullability)?.nullability == KtTypeNullability.NULLABLE
+
 val KtType.isUnit: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.UNIT)
+val KtType.isInt: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.INT)
+val KtType.isLong: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.LONG)
+val KtType.isShort: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.SHORT)
+val KtType.isByte: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.BYTE)
+val KtType.isFloat: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.FLOAT)
+val KtType.isDouble: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.DOUBLE)
+val KtType.isChar: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.CHAR)
+val KtType.isBoolean: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.BOOLEAN)
+val KtType.isString: Boolean get() = isClassTypeWithClassId(DefaultTypeClassIds.STRING)
+
+val KtType.isUInt: Boolean get() = isClassTypeWithClassId(StandardNames.FqNames.uInt)
+val KtType.isULong: Boolean get() = isClassTypeWithClassId(StandardNames.FqNames.uLong)
+val KtType.isUShort: Boolean get() = isClassTypeWithClassId(StandardNames.FqNames.uShort)
+val KtType.isUByte: Boolean get() = isClassTypeWithClassId(StandardNames.FqNames.uByte)
```

Comment:
I know, that the first `isUnit` function was added by me here, but could you please move all of those functions into some `KtAnalysisSessionComponent`. Probably, it's better to introduce a new one, smth like `KtTypeInfoProvider`. The general approach here if we have a non-local function which operates on `ValidityTokenOwner`, it should be  introduced in some `KtAnalysisSessionComponent` it you think it may be useful for others like  KtType.isUnit/isMarkedNullable/isPrimitive. If the function seems to be not usefull outside your scope, it should have explicit receiver of type `KtAnalysisSessoin` even it it is not inside. This is needed to ensure that when we call that function we are always inside some `analyse` call. Eg,  `val KtType.defaultInitializer: String?` should be rewritten to ` fun KtAnalysisSessoin.getDefaultInitializerForType(type: KtType): String?`

Reply:
OK, I will also move everything in this file to `TypeInfoProvider`. In addition, it seems there is no need to delegate the implementation to FIR. Default implementations inside the interface is sufficient. But let me know if you think otherwise.

Reply:
The functions like ` val KtType.isInt: Boolean` are simple and do not use fir, so it's okey to leave their implementation in mixin